account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -69,15 +72,35 @@ def import_file(self):
 
     @api.model
     def _import_file(self, data_file):
-        """ Create bank statement(s) from file
-        """
+        """ Create bank statement(s) from file."""
         # The appropriate implementation module returns the required data
-        currency_code, account_number, stmts_vals = self._parse_file(data_file)
-        # Check raw data
-        self._check_parsed_data(stmts_vals)
+        statement_ids = []
+        notifications = []
+        statements = self._parse_file(data_file)
```

Comment:
I fear those changes might come back to haunt us. While it's true everything might change in upstream tomorrow, the code is at it is today and I think we should stick to it's API. Or change upstream, and then adapt to that.

How about: Leave signatures as they are, but check keys `%currency_code` and `%account_number` (or whatever other prefix makes it unlikely for this to be a column name) in the statement dict which override the according value from the tuple.